feat(STN-58)  foundational css all themes

silverli opened the pull request:
# READY FOR REVIEW

## JIRA Issue
https://sparkbox.atlassian.net/browse/STN-58

# Summary
Creates the foundational CSS for humsci_basic theme, which will then be utilized in child subthemes. Also, uses ITCSS structure to organize our folder.

- adds normalize, utilities, mixins, variables, flex-grid, and some components from Decanter
- also adds z-index variables
- adds function to get different color maps for each theme
- adds sass-true tests for new function
- adds fonts and colors for the basis of the colorful theme

# Steps to Test
1. See questions I added in the comments.
1. Run `npm run watch` in the humsci_basic folder, add a utility from decanter in `_base.css` to see if it compiles.
2. Run `npm run test` and make sure all tests and linting pass

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

Commit:
feat(STN-58): Integrate foundational mixins, functions, variables and styles from Decanter 6.0

silverli commented on docroot/themes/humsci/humsci_basic/package.json:
```diff
@@ -17,7 +17,7 @@
   "dependencies": {
     "bourbon": "^5.1.0",
-    "decanter": "^5.1.0"
+    "decanter": "^6.0.0-alpha.1"
```
need to further analyze version 6 when it is released and what new things we might want to include from it (should be happening today)

silverli replied:
release notes here https://github.com/SU-SWS/decanter/releases/tag/6.0.0

silverli replied:
lots of general cleanup occurred, as well as adding two new components: secondary nav and card icon

silverli commented on docroot/themes/humsci/humsci_basic/src/scss/_base.scss:
```diff
@@ -1,8 +1,18 @@
-// @use 'decanter/core/src/scss/components/brand-bar/index';
-@import 'decanter/core/src/scss/decanter-no-markup';
+@import
+  'decanter/core/src/scss/decanter-no-markup',
```
note to myself: no-markup brings in everything in the utilities folder, like variables, mixins, functions, and placeholders that can be extended

silverli commented on docroot/themes/humsci/humsci_basic/src/scss/_base.scss:
```diff
@@ -1,8 +1,18 @@
+  'decanter/core/src/scss/core/base',
+  'decanter/core/src/scss/core/helpers',
+  'decanter/core/src/scss/elements',
+  'decanter/core/src/scss/core/flex-grid'
```
all this does is generate classes that can be applied to html, I think all the relevant flex and grid mixins are imported with decanter-no-markup

silverli commented on docroot/themes/humsci/humsci_basic/src/scss/_base.scss:
```diff
@@ -1,8 +1,18 @@
+  'decanter/core/src/scss/core/helpers',
+  'decanter/core/src/scss/elements',
+  'decanter/core/src/scss/core/flex-grid'
+;
```
we need to import the font imports too: `core/src/scss/core/fonts/_fonts.scss`

silverli commented on docroot/themes/humsci/humsci_basic/src/scss/_base.scss:
```diff
@@ -1,8 +1,18 @@
+  'decanter/core/src/scss/decanter-no-markup',
+  'decanter/core/src/scss/core/normalize',
+  'decanter/core/src/scss/core/base',
+  'decanter/core/src/scss/core/helpers',
```
hm this just brings in classes that use a mixin, not sure we need it

silverli commented on docroot/themes/humsci/humsci_basic/src/scss/_base.scss:
```diff
@@ -1,8 +1,18 @@
+  'decanter/core/src/scss/core/normalize',
+  'decanter/core/src/scss/core/base',
+  'decanter/core/src/scss/core/helpers',
+  'decanter/core/src/scss/elements',
```
this includes specific, overriding styles for h1-h6, lists, tables, and other certain elements. we can keep this for now, but might want to see how much we have to override to start building the colorful theme

silverli commented on docroot/themes/humsci/humsci_basic/src/scss/_base.scss:
```diff
@@ -1,8 +1,18 @@
 
-// Decanter's use of global variables won't allow us to import with @use
-// @use 'decanter/core/src/scss/components/brand-bar/index';
-@import 'decanter/core/src/scss/decanter-no-markup';
```
we can bring in these as well, I think since every site needs to have them:

- brand bar https://github.com/SU-SWS/decanter/tree/v6/core/src/scss/components/brand-bar
- lockup https://github.com/SU-SWS/decanter/tree/v6/core/src/scss/components/lockup
- footer things https://github.com/SU-SWS/decanter/tree/v6/core/src/scss/components/global-footer and https://github.com/SU-SWS/decanter/tree/v6/core/src/scss/components/local-footer

silverli replied:
we might want to consider using the navigation too 
- https://github.com/SU-SWS/decanter/tree/6.0.0/core/src/scss/components/secondary-nav 
- https://github.com/SU-SWS/decanter/tree/6.0.0/core/src/scss/components/main-nav
- https://github.com/SU-SWS/decanter/tree/6.0.0/core/src/scss/components/nav-toggle

silverli replied:
this is accompanied by some JS we can import too if needed

silverli replied:
I'd like feedback on what @nrambeck or @bryanbraun thinks about bringing these in

Commit:
feat(STN-58): add Decanter 6.0.0 and fonts, and other brand specific components

pookmish commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr441-epr3nr91bto4ugvvycee3fw1pi4iuuqj.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5e0f69e59a3cd568bb3cda4c

Visual Diffs:
- https://dashboard.tugboat.qa/5e0f69e59a3cd568bb3cda4c/visualdiffs

Commit:
feat(STN-58): add zindex variables

pookmish commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr441-epr3nr91bto4ugvvycee3fw1pi4iuuqj.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5e0f69e59a3cd568bb3cda4c

Visual Diffs:
- https://dashboard.tugboat.qa/5e0f69e59a3cd568bb3cda4c/visualdiffs

Commit:
feat(STN-58): reorganize into ITCSS structure and adds color maps and functions

- adds map for colorful theme, leaves airy and traditional blank
- adds functions for obtaining theme-color() based on a certain key given
- adds sass-true tests for new function
- refactors sass true spec file for dart-sass

pookmish commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr441-epr3nr91bto4ugvvycee3fw1pi4iuuqj.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5e0f69e59a3cd568bb3cda4c

Visual Diffs:
- https://dashboard.tugboat.qa/5e0f69e59a3cd568bb3cda4c/visualdiffs

Commit:
feat(STN-58): completing ITCSS imports and matching decanter import load order

pookmish commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr441-epr3nr91bto4ugvvycee3fw1pi4iuuqj.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5e0f69e59a3cd568bb3cda4c

Visual Diffs:
- https://dashboard.tugboat.qa/5e0f69e59a3cd568bb3cda4c/visualdiffs

bryanbraun commented on docroot/themes/humsci/humsci_basic/src/scss/tools/_functions.colors.scss:
```diff
@@ -0,0 +1,32 @@
+// Theme Color Function
+// Maps over $humsci-basic--theme-colors array
+// with matching keys to the theme and color requested
+// Params: ('string'), e.g. theme-colors('teal')
```
Should this say `theme-color('teal')` (singular)? 

Maybe you can walk me through the use case for these color functions?

bryanbraun commented on docroot/themes/humsci/humsci_basic/package.json:
```diff
@@ -17,7 +17,7 @@
   "dependencies": {
     "bourbon": "^5.1.0",
-    "decanter": "^5.1.0"
+    "decanter": "^6.0.0"
```
I'm glad we did this. 👍

bryanbraun commented on docroot/themes/humsci/humsci_basic/src/scss/_base.scss:
```diff
@@ -1,18 +1,16 @@
 html {
   font-size: 1em;
+  // how to get the Decanter Stanford Colors with a sass function
+  color: color(brand);
```
I'm not seeing where `color()` is defined...

bryanbraun replied:
Update: it's from decanter.

bryanbraun commented on docroot/themes/humsci/humsci_basic/src/scss/settings/_variables.fonts.scss:
```diff
@@ -0,0 +1,2 @@
+// Colorful Theme Fonts
+$colorful-font--sans: $su-font-sans;
```
I like this. 👍

bryanbraun commented on docroot/themes/humsci/humsci_basic/src/scss/settings/_variables.theme.colors.scss:
```diff
@@ -0,0 +1,15 @@
+// I cannott call these teal, purple, etc because sass tries to parse it beforehand
+// All named colors are reserved values in Sass and will be turned into hex values regardless of placement and use.
+// so I prefixed everything with "su-"
```
On second thought, `su-` is used as a decanter prefix, so using it here might create some confusion. Maybe it's better to just do `color-teal`, `color-mint`, etc.

bryanbraun commented on docroot/themes/humsci/humsci_basic/src/scss/_main.scss:
```diff
@@ -1,4 +1,90 @@
+  'decanter/core/src/scss/core/helpers',
+  'decanter/core/src/scss/core/flex-grid',
+  'decanter/core/src/scss/core/base',
+  'decanter/core/src/scss/elements',
```
Per your previous comments:

- `decanter/core/src/scss/core/flex-grid`: Lets leave this out and pull it in later if we need it. It looks like we aren't really using any of these classes in our existing templates.
- `decanter/core/src/scss/core/helpers`: I'm ok with keeping these helpers. There are only a few and they seem fairly useful (for screenreaders, etc).
- `decanter/core/src/scss/elements`: I agree, let's include this for now, but if we find that we're doing tons of overrides, we should consider excluding it. Maybe we add a code comment that communicates this intent?

bryanbraun reviewed (changes requested):
I make just a few minor requests in my comments above but by and large this looks great.

Commit:
feat(STN-58): pr review comments

pookmish commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr441-epr3nr91bto4ugvvycee3fw1pi4iuuqj.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/5e0f69e59a3cd568bb3cda4c

Visual Diffs:
- https://dashboard.tugboat.qa/5e0f69e59a3cd568bb3cda4c/visualdiffs

bryanbraun commented:
There's still an open question about behat tests failing that we'll want to look into before merging `new-themes` into the release branch, but we can move this forward for now.
